code/mathematical_algorithms/src/add_polynomials/add_polynomials.cpp

```diff
@@ -57,8 +63,8 @@ void polynomial::insert_term(int c, int p) {
   }
 }
 
-void polynomial::print() {
-  term* t = head;
+void Polynomial::print() {
```

Comment:
This function is useless, TBH. Not every application uses a console, so it would be better if you returned the polynomial as a string. If you really want a print function, then overload the `<<` operator so it's not just restricted to stdout.